cmd/commands/project.go

```diff
@@ -324,3 +327,112 @@ var getInstallationNamespace = func(repofs fs.FS) (string, error) {
 
 	return a.Spec.Destination.Namespace, nil
 }
+
+type (
+	ProjectListOptions struct {
+		FS           fs.FS
+		CloneOptions *git.CloneOptions
+	}
+)
+
+func NewProjectListCommand() *cobra.Command {
+	var (
+		cloneOpts *git.CloneOptions
+	)
+
+	cmd := &cobra.Command{
+		Use:   "list ",
+		Short: "Lists all the projects on a git repository",
+		Example: util.Doc(`
+# To run this command you need to create a personal access token for your git provider,
+# and have a bootstrapped GitOps repository, and provide them using:
+	
+		export GIT_TOKEN=<token>
+		export GIT_REPO=<repo_url>
+
+# or with the flags:
+	
+		--token <token> --repo <repo_url>
+		
+# Lists projects
+	
+	<BIN> project list
+`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			return RunProjectList(cmd.Context(), &ProjectListOptions{
+				FS:           fs.Create(memfs.New()),
+				CloneOptions: cloneOpts,
+			})
+		},
+	}
+
+	cloneOpts, err := git.AddFlags(cmd)
+	die(err)
+
+	return cmd
+}
+
+func RunProjectList(ctx context.Context, opts *ProjectListOptions) error {
+
+	var (
+		err error
+	)
+	log.G().WithFields(log.Fields{
+		"repoURL":  opts.CloneOptions.URL,
+		"revision": opts.CloneOptions.Revision,
+	}).Debug("starting with options: ")
+
+	log.G().Infof("using revision: \"%s\", installation path: \"%s\"", opts.CloneOptions.Revision, opts.FS.Root())
+
+	// clone repo
+	log.G().Infof("cloning git repository: %s", opts.CloneOptions.URL)
+	_, opts.FS, err = opts.CloneOptions.Clone(ctx, opts.FS)
+	if err != nil {
+		return err
+	}
+	if !opts.FS.ExistsOrDie(store.Default.BootsrtrapDir) {
+		log.G().Fatalf("Bootstrap folder not found, please execute `repo bootstrap --installation-path %s` command", opts.FS.Root())
+	}
+
+	matches, err := billyUtils.Glob(opts.FS, "/projects/*.yaml")
+	if err != nil {
+		return err
+	}
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 2, ' ', 0)
+	_, _ = fmt.Fprintf(w, "NAME\tNAMESPACE\tCLUSTER\t\n")
+
+	for _, name := range matches {
+		proj, err := getProjectInfoFromFile(opts.FS, name)
+		if err != nil {
+			return err
+		}
+		fmt.Fprintf(w, "%s\t%s\t%s\n", proj.Name, proj.Namespace, proj.ClusterName)
+
+	}
+	w.Flush()
+	return nil
+}
+
+func getProjectInfoFromFile(fs fs.FS, name string) (*argocdv1alpha1.AppProject, error) {
```

Comment:
I think this should return 3 types `(*argocdv1alpha1.AppProject, *ApplicationSet, error)` even if we don't need the application set now